### PR TITLE
Multiselect rework

### PR DESCRIPTION
**Pull request by AlphaLima**
Complete rework of mutliselect for:
-Flarm
-Oracle
-Team
-Own Position
alternative threadsafe UTF8 Pictorials 
all changes with Option switch in Options.h

Example:
multisel
multisel

### CODE REVIEW

**Comment by pventafridda** [comment]
nice pictorials!!!!


In data 14 Novembre 2017 10:49:17 PM Ulli Heynen <notifications@github.com> 
ha scritto:

**Comment by AlphaLima** [comment]
UTF8 is not the point of this commit, it was just an alternative in case the FLARM vector pictorials are not threadsafe.
This commit it is about Traffic Multiselect. Is it now OK now?

**Comment by pventafridda** [comment]
we need to advance to 6.3a after release of 6.2 before merging ..


In data 16 Novembre 2017 3:18:38 PM Ulli Heynen <notifications@github.com> 
ha scritto:

**Comment by AlphaLima** [comment]
of course! no need to hurry.

**Comment by AlphaLima** [comment]
OK, now without UTF8 characters.

File: Common/Source/Dialogs/dlgMultiSelectList.cpp [hunk 1]
```diff
@@ -533,15 +568,7 @@ FLARM_TRAFFIC* pFlarm;
...
-            _stprintf(Comment,_T("⛳"));
-
-            Surface.SetBackgroundTransparent();
-            OldFont =  Surface.SelectObject(LK8PanelBigFont);
-
-            Surface.SetTextColor(RGB_BLUE);
-            xtext = Surface.GetTextWidth(Comment);
-            Surface.DrawText(rc.left +(rc.right-rc.left-xtext)/2 , DLGSCALE(2), Comment);
-            Surface.SelectObject(OldFont);
+            UTF8Pictorial( Surface,  rc, (TCHAR*)_T("⛳"),LKColor(90,150,90));
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgMultiSelectList.cpp]: wrong cast from "const TCHAR*" to "TCHAR*"

File: Common/Source/Dialogs/dlgMultiSelectList.cpp [hunk 2]
```diff
@@ -554,22 +581,14 @@ FLARM_TRAFFIC* pFlarm;
...
             else
               _stprintf(text2,_T("%s"),   gettext(_T("_@M1690_"))); //_@M1690_ "THE LK8000 ORACLE"
-            _stprintf(Comment,_T("♛"));
-            Surface.SetBackgroundTransparent();
-            OldFont =  Surface.SelectObject(LK8PanelBigFont);
-            Surface.SetTextColor(RGB_DARKGREEN);
-            xtext = Surface.GetTextWidth(Comment);
-            Surface.DrawText(rc.left +(rc.right-rc.left-xtext)/2 , DLGSCALE(2), Comment);
-            Surface.SelectObject(OldFont);
+            UTF8Pictorial( Surface,  rc, (TCHAR*)_T("♛"),RGB_BLUE);
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgMultiSelectList.cpp]: wrong cast from "const TCHAR*" to "TCHAR*"

File: Common/Source/Dialogs/dlgMultiSelectList.cpp [hunk 3]
```diff
@@ -569,7 +587,12 @@ FLARM_TRAFFIC* pFlarm;
         case IM_TEAM:
             _stprintf(text1,_T("%s:"),gettext(_T("_@M700_"))); //_@M700_ "Team code"
             _stprintf(text2,_T("%s"), CALCULATED_INFO.OwnTeamCode );
-            UTF8Pictorial( Surface,  rc, (TCHAR*)_T("⛳"), RGB_ORANGE);
+            ShowTextEntries(Surface, rc,  text1, text2);
+#ifdef KOBO
+              UTF8Pictorial( Surface,  rc, (TCHAR*)_T("♥"), RGB_BLACK);
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgMultiSelectList.cpp]: wrong cast from "const TCHAR*" to "TCHAR*"

File: Common/Source/Dialogs/dlgMultiSelectList.cpp [hunk 4]
```diff
@@ -569,7 +587,12 @@ FLARM_TRAFFIC* pFlarm;
...
+#else
+              UTF8Pictorial( Surface,  rc, (TCHAR*)_T("⛳"), RGB_ORANGE);
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgMultiSelectList.cpp]: wrong cast from "const TCHAR*" to "TCHAR*"

File: Common/Source/Dialogs/dlgMultiSelectList.cpp [hunk 5]
```diff
@@ -240,13 +288,217 @@ void dlgAddMultiSelectListItem(long* pNew, int Idx, char type, double Distance)
...
+void DrawUTF8FlarmPicto(LKSurface& Surface, const RECT& rc, FLARM_TRAFFIC* pTraf)
+{
+if (pTraf == NULL) return;
+double fFact = fabs(pTraf->Average30s + 5.0) /10.0;
+if(fFact > 1.0 ) fFact = 1.0; else
+  if(fFact < 0.0 ) fFact = 0.0;
+LKColor BaseColor = RGB_GREEN ;
+extern  LKColor MixColors(const LKColor& Color2, double fFact1) ;
+BaseColor =  BaseColor.MixColors( RGB_BLUE, fFact);
+if(pTraf->Status == LKT_GHOST)  UTF8Pictorial( Surface,  rc,(TCHAR*) _T("■") ,BaseColor); else
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgMultiSelectList.cpp]: wrong cast from "const TCHAR*" to "TCHAR*"

same for 2 next line

File: Common/Source/Dialogs/dlgMultiSelectList.cpp [hunk 6]
```diff
@@ -297,157 +552,147 @@ static void OnMultiSelectListPaintListItem(WindowControl * Sender, LKSurface& Su
...
+             * IM_ORACLE
+             ************************************************************************************************/
+        case IM_ORACLE:
+            _stprintf(text1,_T("%s"),gettext(_T("_@M2058_"))); //_@M2058_ "Oracle"
+            if(Elements[i].iIdx >= 0)
+              _stprintf(text2,_T("%s: %s"),    gettext(_T("_@M456_")), WayPointList[Elements[i].iIdx].Name);// _@M456_ "Near"
+            else
+              _stprintf(text2,_T("%s"),   gettext(_T("_@M1690_"))); //_@M1690_ "THE LK8000 ORACLE"
+            ShowTextEntries(Surface, rc,  text1, text2);
+            UTF8Pictorial( Surface,  rc, (TCHAR*)_T("?"),RGB_BLUE);
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgMultiSelectList.cpp]: wrong cast from "const TCHAR*" to "TCHAR*"

**Pull request closed without merging**